src/dftbp/timedep/transcharges.F90

```diff
@@ -113,18 +135,156 @@ subroutine TTransCharges_init(this, iAtomStart, sTimesGrndEigVecs, grndEigVecs,
     !> index array for single particle excitations that are included
     integer, intent(in) :: win(:)
 
-    integer :: ia, ij, ii, jj, kk, ab, aa, bb
+    integer :: ia, ii, jj, ij, kk, ab, aa, bb, ss
     logical :: updwn
 
-    this%nTransitions = nTrans
-    this%nAtom = size(iAtomStart) - 1
+  #:if WITH_SCALAPACK
+    
+    integer :: iSpin, iAtom, nSpin, nOrbs, nLocRow, nLocCol
+    integer :: desc(DLEN_), iLoc, mLoc, iGlb, mGlb, jLoc, jGlb
+    real(dp), allocatable :: maskMat(:,:), workLocal(:,:), workGlobal(:,:,:)
+
+  #:endif
+    
+    this%nTransitions = nTrans  
+    this%nAtom = size(denseDesc%iAtomStart) - 1
     this%nMatUp = nMatUp
     this%nMatUpOccOcc = nXooUD(1)
     this%nMatUpVirVir = nXvvUD(1)
 
+  #:if WITH_SCALAPACK
+
+    nSpin = size(grndEigVecs, dim=3)
+     
+    if (tStoreSame) then
+
+      @:ASSERT(tStoreOccVir)
+      if (tFirstCall) then
+        @:ASSERT(.not.allocated(this%qCacheOccOcc))
+        allocate(this%qCacheOccOcc(this%nAtom, sum(nXooUD)))
+        @:ASSERT(.not.allocated(this%qCacheVirVir))
+        allocate(this%qCacheVirVir(this%nAtom, sum(nXvvUD)))
+      end if
+      
+    else
+      
+      this%tCacheChargesSame = .false.
+      
+    end if
+    
     if (tStoreOccVir) then
+      
+      if (tFirstCall) then
+        @:ASSERT(.not.allocated(this%qCacheOccVir))
+      else
+        ! A simple deallocate gives me an error, why is this%qCacheOccVir
```

Comment:
Ans. It's due to the second call to TTransCharges_init from after nxov_rd is redefined due to oscillator windowing in LinRespGrad_old()

Reply:
ok!